src/devices/openxrheadset/OpenXrHeadset.cpp

```diff
@@ -452,75 +458,76 @@ bool yarp::dev::OpenXrHeadset::threadInit()
             }
             slide.layer.setVisibility(IOpenXrQuadLayer::Visibility::BOTH_EYES);
             slide.layer.setDimensions(slide.width, slide.height);
-            slide.layer.setPosition({slide.x, slide.y, slide.z});
+            slide.layer.setPosition({ slide.x, slide.y, slide.z });
             slide.layer.setImage(slide.options.initialSlide);
         }
-    }
-
-    for (size_t i = 0; i < 10 && m_openXrInterface.isRunning(); ++i)
-    {
-        run(); //dry run. This is to make sure that the number of buttons is correctly retrieved by the JoypadControlServer
-        yarp::os::Time::delay(this->getPeriod());
-    }
-
-    {
-        std::lock_guard<std::mutex> lock(m_mutex);
 
         this->yarp().attachAsServer(this->m_rpcPort);
-        if(!m_rpcPort.open(m_rpcPortName))
+        if (!m_rpcPort.open(m_rpcPortName))
         {
             yCError(OPENXRHEADSET) << "Could not open" << m_rpcPortName << " RPC port.";
             return false;
         }
     }
 
+    {
+        std::lock_guard<std::mutex> lock(m_joypadServerMutex);
+        m_thisDevice.give(this, false);
```

Comment:
Minor:

```suggestion
        m_thisDevice.give(this, /*own=*/false);
```

See https://abseil.io/tips/94 for rationale.

Reply:
Whoops, I was too fast in merging

Reply:
No problem, I indeed commented after approving.

Reply:
I have opened https://github.com/ami-iit/yarp-device-openxrheadset/pull/51